File: Programmers/이진변환반복하기.py
def binary(n):
    n_binary = ''
    while n != 1:
        n_binary = str(n % 2) + n_binary
        n = n // 2
    return '1' + n_binary

def solution(s):
    changed, zero_removed = 0, 0
    
    while True: 
        if s == '1':
            return [changed, zero_removed]
        zero_removed += s.count('0')
        s = s.replace('0','')
        s = binary(len(s))
        changed += 1

File: JavaScript/이진변환반복하기.js
const binary = (num) => {
    let result = '';
    while (num > 1) {
        result += String(num % 2);
        num = Math.floor(num / 2);
    }
    return '1' + result;
}

function solution(s) {
    let count = 0, zeroes = 0;
    while (s !== '1') {
        let ones = 0;
        for (let i = 0; i < s.length; i++) {
            if (s[i] === '0') zeroes += 1;
            else ones += 1;
        }
        s = binary(ones);
        count += 1;
    }
    return [count, zeroes];
}
